src/main/java/com/movie_collection/bll/services/MovieService.java

```diff
@@ -1,12 +1,48 @@
 package com.movie_collection.bll.services;
 
+import com.google.inject.Inject;
+import com.movie_collection.be.Movie;
 import com.movie_collection.bll.services.interfaces.IMovieService;
+import com.movie_collection.dal.interfaces.IMovieDAO;
 
+import java.sql.SQLException;
 import java.util.List;
 
 public class MovieService implements IMovieService {
+    private final IMovieDAO movieDAO;
+
+    @Inject
```

Comment:
sweet use of injection 
do u think it makes sense to have it like this ? 
if its right implementation of DAO

Reply:
I will take your advice and recommendations on this since you have more experience and did more research

Reply:
just making sure we are understand that if they will ask any of us we know why we did it 
and maybe even why are we using this design pattern for our application :)

Reply:
Alright, I think it will be best for you to explain it to us further just to be sure